### ISSUE TITLE
launchdarkly_feature_flag_environment: Support setting no value_type when providing empty values for clauses

### ISSUE THREAD

**Issue by nc-wittj** [launchdarkly/terraform-provider-launchdarkly#102]
In our project, we sometimes like to use the `launchdarkly_feature_flag_environment` resource to set up rules with clauses containing no values, so they are ready to fill in once we want to rollout the feature flag. 

Previously, we were on `v1.5.0`, and providing empty `values` worked fine. Now, on `v2.7.0`, I'm noticing that each time I run `terraform apply`, it tries to set `value_type = "string"` for clauses with no values. My guess is on the API clears out the `value_type` when there are no values, triggering the Terraform Provider to try to set it again next time via the defaulting logic. 

Would it be possible to add a way to set `value_type` to some `null` value when our values are empty, so we aren't caught in this cycle? For now, I think my work around will be to conditionally not create the rule when all the `values` are empty. But that doesn't tell the story as nicely when viewed in the UI.

**Comment by lucywyman** [comment]
Hm...we use the `value_type` to determine whether we need to convert the string value that's provided into a boolean or number before sending it to the API, so setting it to `null` (or `'any'`) would be effectively the same as setting it to the default `string`. We could in theory not explicitly set the default in the TF plugin schema so that`value_type` is omitted from the TF state file if it's not provided, and then under the hood use `string` as a default. But someone could have the opposite problem where they aren't sure why their boolean value is being treated as a string and it'd be hard to debug...

I guess I'm wondering why it's a problem that `value_type=string` is set? It's only used internally to do the checking, and isn't included in API calls and shouldn't (as far as I can see at least) affect the actual feature flag that gets created. Is it just confusing that it's there?

**Comment by nc-wittj** [comment]
> I guess I'm wondering why it's a problem that `value_type=string` is set?

The issue on our end is that _every_ time we run a `terraform apply`, it ends up trying to set the `value_type` _again_. It doesn't matter how many times we apply. And when Terraform updates `value_type`, it triggers a notification in Slack. For us, that means over 30 notifications each time we apply, creating unnecessary noise.

If you added some logic so it didn't try to keep setting the `value_type` if `values` is empty, I think that'd fix the issue for us. I'm fine with `value_type` being set; it's just an issue when we provide an empty `values` array.

Here's a simple project that reproduces the issue:
```terraform
terraform {
  required_version = "~> 1.2"

  required_providers {
    launchdarkly = {
      source = "launchdarkly/launchdarkly"
    }
  }
}

variable "access_token" {
  type = string
}

provider "launchdarkly" {
  access_token = var.access_token
}

resource "launchdarkly_project" "project" {
  key  = "issue-102"
  name = "Reproducing https://github.com/launchdarkly/terraform-provider-launchdarkly/issues/102"

  environments {
    name  = "issue-102"
    key   = "issue-102"
    color = "000000"
  }
}

resource "launchdarkly_feature_flag" "flag" {
  project_key = launchdarkly_project.project.key
  name        = "Example Flag"
  key         = "example"

  variation_type = "boolean"
}

resource "launchdarkly_feature_flag_environment" "env" {
  env_key = launchdarkly_project.project.environments[0].key
  flag_id = launchdarkly_feature_flag.flag.id

  # We setup a rule so we are ready, even if we don't yet know the values.
  # For example, the users are still yet to be created, but we want to record
  # in code that we intend to rollout to a small set of users first.
  rules {
    clauses {
      attribute = "UserID"
      op        = "in"

      # Since values is empty, every time we run `terraform apply` it tries to update the `value_type`, which
      # the API subsequently clears. I would expect there to be "no changes" after the first `terraform apply`.
      values = []
    }
  }

  off_variation = 1

  fallthrough {
    variation = 0
  }
}
``` 

After you apply the first time, you'll notice subsequent applies will result in:
```terraform
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # launchdarkly_feature_flag_environment.env will be updated in-place
  ~ resource "launchdarkly_feature_flag_environment" "env" {
        id            = "issue-102/issue-102/example"
        # (5 unchanged attributes hidden)


      ~ rules {
            # (2 unchanged attributes hidden)

          ~ clauses {
              + value_type = "string"
                # (4 unchanged attributes hidden)
            }
        }
        # (1 unchanged block hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

I would expect no changes to be necessary.

**Comment by lucywyman** [comment]
Ah, ok, I can see how that's an issue. I'll make a ticket to have the default get set only if `values` is not empty :+1:

**Comment by lucywyman** [comment]
This should be fixed in 2.9.2 (latest is currently 2.9.3)!
